zcash_client_sqlite/src/lib.rs

```diff
@@ -252,14 +266,14 @@ impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletRead for W
         account_id: Self::AccountId,
         seed: &SecretVec<u8>,
     ) -> Result<bool, Self::Error> {
-        self.get_unified_full_viewing_keys().and_then(|keys| {
-            keys.get(&account_id).map_or(Ok(false), |ufvk| {
+        if let Some(account) = self.get_account(account_id)? {
+            if let Account::Zip32(HdSeedAccount(_, account_index, ufvk)) = account {
```

Comment:
We now have the seed fingerprint, so we should check that before deriving `usk`, both to ensure that the fingerprint matches, and because checking the fingerprint is cheaper than deriving `usk`.